src/main/resources/diagram_git_starship.puml

```diff
@@ -2,132 +2,187 @@
 
 package rape.brutal.gitstarship {
 
-    package ships {
+    class Universe {
+        -int kills;
+        -int moneySpends;
+        +void startUniverse();
+        +void winState();
+        +void loseState();
+        +void getStats();
+    }
 
-        Starship <|-- GitStarship
-        Starship <|-- PirateStarship
+    class Storage {
+        -HashMap<String, IStorageable> items;
+        -int capacity;
+        +void putItem(IStorageable iStorageable);
+        +void getItem(String itemName);
+    }
 
-        abstract class Starship {
-            #Engine engine;
-            #Hull hull;
-            #Storage<IStoreable> storage;
+    package trade {
 
-            +boolean putInStorage(IStoreable iStoreable);
-            +IStoreable getFromStorage(String iStoreableName);
+        abstract class Trader {
+            -int cash;
+            -Storage storage;
+        }
 
-            -void flyTo(IReachable iReachable);
+        class AverageTrader extends Trader {
+            +boolean sell(int cash, String itemName);
+            +boolean buy(int cash, String itemName);
+        }
 
-            +void trade(Trader trader);
+        class Mechanic extends Trader {
+            +boolean repair(int cash, String itemName);
+            +boolean upgrade(int cash, String itemName);
+        }
 
-            +void setDamage(int damage);
-            +int getHitPoints();
+    }
 
-            +void attack(IAttackable iAttackable);
-        }
+    package events {
 
-        class GitStarship {
-            -Railgun railgun;
-            -Turret rightTurret;
-            -Turret leftTurret;
-        }
+        +abstract class Event {
+            #Universe universe;
+            #Location location;
+            #GitStarship gitStarship;
 
-        class PirateStarship {
-            -Turret rightTurret;
+            +void Event(Universe universe, Location location, GitStarship gitStarship);
+            +void start(GitStarship gitStarship);
```

Comment:
If `gitStarship` assigned via the constructor, it is a unnesessary to pass it through `start()` parameter.